barretenberg/cpp/src/barretenberg/client_ivc/client_ivc.cpp

```diff
@@ -104,75 +104,62 @@ ClientIVC::perform_recursive_verification_and_databus_consistency_checks(
 {
     using MergeCommitments = Goblin::MergeRecursiveVerifier::InputCommitments;
 
-    // Witness commitments and public inputs corresponding to the incoming instance
-    WitnessCommitments witness_commitments;
     // The pairing points produced by the verification of the decider proof
     PairingPoints decider_pairing_points;
-    std::vector<StdlibFF> public_inputs;
 
     // Input commitments to be passed to the merge recursive verification
-    MergeCommitments merge_commitments;
-    merge_commitments.T_prev_commitments = T_prev_commitments;
+    MergeCommitments merge_commitments{ .T_prev_commitments = T_prev_commitments };
+
+    auto verifier_instance =
```

Comment:
We always have an incoming `verifier_instance` regardless of the type of recursion being performed. The witness_commitments and public_inputs are uniformly extracted from this after recursion. This also makes it clearer what witness commitments and pub inputs were talking about